src/node_i18n.cc

```diff
@@ -761,25 +751,26 @@ static int GetColumnWidth(UChar32 codepoint,
     case U_EA_HALFWIDTH:
     case U_EA_NARROW:
     default:
+      const auto zero_width_mask = U_GC_CC_MASK |  // C0/C1 control code
+                                  U_GC_CF_MASK |  // Format control character
+                                  U_GC_ME_MASK |  // Enclosing mark
+                                  U_GC_MN_MASK;   // Nonspacing mark
+      if (codepoint != 0x00AD &&  // SOFT HYPHEN is Cf but not zero-width
+          ((U_MASK(u_charType(codepoint)) & zero_width_mask) ||
+          u_hasBinaryProperty(codepoint, UCHAR_EMOJI_MODIFIER))) {
+        return 0;
+      }
       return 1;
   }
 }
 
 // Returns the column width for the given String.
 static void GetStringWidth(const FunctionCallbackInfo<Value>& args) {
   Environment* env = Environment::GetCurrent(args);
-  if (args.Length() < 1)
-    return;
+  CHECK(args[0]->IsString());
 
   bool ambiguous_as_full_width = args[1]->IsTrue();
-  bool expand_emoji_sequence = args[2]->IsTrue();
-
-  if (args[0]->IsNumber()) {
-    uint32_t val;
-    if (!args[0]->Uint32Value(env->context()).To(&val)) return;
-    args.GetReturnValue().Set(GetColumnWidth(val, ambiguous_as_full_width));
-    return;
-  }
+  bool expand_emoji_sequence = !args[2]->IsBoolean() || args[2]->IsTrue();
```

Comment:
Can you explain why the expansion behaviour is preferable? The commit message for this change seems to contradict the actual change here:

> Most terminals have emoji support by now and visualize them as full width characters.

Reply:
Good catch, the commit message is indeed wrong. I'll fix that soon. Most potential receivers do not seem to joined emojis and I could not find a reliable way to detect if the receiver is capable to handle them or not. So falling back to visualizing them as individual characters still seems best.
We should probably expose these options to the user. I expect they would know best if the receiving end supports these or not. I added a TODO entry for that.

Reply:
What makes it even worse is that some receivers can support joining some emoji sequences and not others and can be very inconsistent. There's absolutely no way of determining how the sequence will be handled in any given terminal without doing essentially what we have for color support for specific terminals in tty.js. The challenge with exposing options to the user is that if the application is expected to operate in multiple terminals, the user may not have any idea what to expect either, but it's likely the best option we've got.

Reply:
Including a check similar to the color one sounds good to me. But collecting all the necessary data is difficult.

Reply:
I rebased and updated the commit message. PTAL.